### PR TITLE
feat: init commit

### PR DESCRIPTION
**Pull request by leongold**

### CODE REVIEW

File: config.dev.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (orkamara) [inline config.dev.json]: what about `config.prod.json`

File: src/common/types.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  size: number;
+  digest: Digest;
+}
+export interface Layer {
+  config: LayerConfig;
+  blob: Buffer;
+}
+
+export type AnyJson = boolean | number | string | null | JsonArray | JsonMap;
+export interface JsonArray extends Array<AnyJson> {}
```
Comment (orkamara) [inline src/common/types.ts]: `JsonArray` + `JsonMap` - not in use

File: src/app.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import sentry = require("./common/sentry");
+import heathCheck = require("./routes/healthcheck");
+
+import { LayersCache } from "./layers-cache";
+import { getLayer } from "./routes/get-layer";
+import { saveLayer } from "./routes/save-layer";
+
+export const app = express();
+export const layersCache = new LayersCache(
+  config.LAYERS_CACHE_DIR_PATH,
```
Comment (orkamara) [inline src/app.ts]: those 2 configs are not part of the default config?

Comment (orkamara) [reply src/app.ts]: there is not default val for LAYERS_CACHE_DIR_PATH

File: src/routes/save-layer.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import * as express from "express";
+import { layersCache } from "../app";
+import { Logger } from "../common/logger";
+
+const logger = Logger("save-layer");
+
+export async function saveLayer(req: express.Request, res: express.Response) {
+    const digest = req.headers.digest;
```
Comment (orkamara) [inline src/routes/save-layer.ts]: what if `digest` doesn't exist?

File: src/app.ts (hunk 3 above)
Comment (leongold) [reply src/app.ts]: do we need it? shouldn't work without imo

File: src/routes/save-layer.ts (hunk 4 above)
Comment (leongold) [reply src/routes/save-layer.ts]: still need to add error handling/logging in both new routes

**Pull request merged**